training/coach.py

```diff
@@ -307,29 +404,27 @@ def train(self):
                     pos_loss += self.cfg.positive_weights[pos_ind] * (1 - curr_pos_cosine_sim)
                 max_pos_cosine, pos_max_ind = pos_cosine_sim.mean(dim=1).max(dim=0)
                 print(f'\tpos_loss: {pos_loss.item():.3f}, '
-                      f'max_pos: {max_pos_cosine:.3f} for {pos_prompts[pos_max_ind]} ')
+                      f'max_pos: {max_pos_cosine:.3f} for {pos_prompts[pos_max_ind]}')
 
                 neg_prompts = [batch["template"][0].format(token=neg_word) for neg_word in self.cfg.negative_classes]
-                if len(neg_prompts) > 0:
-                    # Calc distances to negative classes
-                    neg_embeds = self.get_normed_embeds(neg_prompts)
-                    neg_cosine_sim = (neg_embeds.detach() @ image_emb_normed.T)
+                if self.cfg.image_feature:
+                    # get feature from data set, image_embs, to do normalize
+                    neg_prompts = [batch["template"][0].format(token=neg_word) for neg_word in negative_classes[batch["template"][0]]]
+                    list_embeds = self.normalize_embeds(torch.stack(image_embs[batch["template"][0]])).detach() if image_embs[batch["template"][0]] else None
```

Comment:
fetch our negative feature by the key like "a photo of a {token}" which will align with prompt of positive image